packages/odyssey-react-mui/src/Icon.tsx

```diff
@@ -14,11 +14,16 @@ import { forwardRef } from "react";
 import type { SvgIconNoChildrenProps } from "./iconDictionary/types";
 import { iconDictionary } from "./iconDictionary";
 
-export interface IconProps extends SvgIconNoChildrenProps {
+export interface IconProps extends Omit<SvgIconNoChildrenProps, "fontSize"> {
   /**
    * Name of the icon to render
    */
   name: keyof typeof iconDictionary;
+
+  /**
+   * Size of the icon
+   */
+  size?: "small" | "medium" | "large";
```

Comment:
I actually don't like default values for sizes like this. I want people to be explicit about it so you can glance at the code and understand what's going on.

I think if we want it optional, `inherit` would make a good optional prop, but if you explicitly define it, it makes sense you'd want small, medium, or large.

But if the design system doesn't want `inherit` then this will be a breaking change where we're forcing folks (including other Odyssey components) to define a size.

I can discuss this at the next UI Guild meeting to see what people like. I dislike hidden props in general. Makes debugging difficult and causes breaking changes if we change the default.

Reply:
This is a great point; I think it makes sense to make `size` a required prop and don't set a default value.

Regarding the breaking change, are you saying we need `inherit` now, or only if we were changing this post-1.0? I'm inclined to not include `inherit` unless we absolutely need it.

PS — I've been really appreciating your feedback on things like this; it's giving me more insight into the strategic/architectural side of engineering and how to think in a future-facing manner rather than just focusing on making a component functional.

Reply:
FWIW our current default size is not `inherit`:

```
    styleOverrides: {
      root: ({ theme }) => ({
        fontSize: `${theme.typography.ui.lineHeight}rem`,
      }),
    },
```

Reply:
Hmm... This will be a breaking change then regardless.

Then I'd say no default value, require the prop, and make it one of those 3 sizes.

Will we never have icons inline in text that we want to match the size of that text? If not, then this change is fine to require `size` and force folks to pass in what they expect.

Reply:
In all current use cases, we want the icons to match the `line-height` of the sibling text, which is why it's the present default.

Reply:
I prefer icon size be set by the size prop instead of the sibling text, if this is not a lot of work.
The current default that inherits the size is 16px in most cases because the line height of text we use most of the times to pair with an icon is 16px.
The new medium size would replace most of our current components.